harmonica/_utils.py

```diff
@@ -0,0 +1,141 @@
+    components of the magnetic vector, respectively.
+
+    The inclination angle is defined as the angle between the magnetic field vector and
+    the horizontal plane:
+
+    .. math::
+
+        Inc = \arctan \frac{- B_u}{\sqrt{B_e^2 + B_n^2}}
+
+    And the declination angle is defined as the azimuth of the projection of the
+    magnetic field vector onto the horizontal plane (starting from the northing
+    direction, positive to the east and negative to the west):
+
+    .. math::
+
+        Dec = \arcsin \frac{B_e}{\sqrt{B_e^2 + B_n^2}}
+
+    .. note::
+        Inclination is measured positive downward from the horizontal plane and
+        declination is measured with respect to North and it is positive east.
+
+
+    Parameters
+    ----------
+    magnetic_e : float or array
+        Easting component of the magnetic vector.
+    magnetic_n : float or array
+        Northing component of the magnetic vector.
+    magnetic_u : float or array
+        Upward component of the magnetic vector.
+    degrees : bool (optional)
+        If True, the angles are returned in degrees.
+        If False, the angles are returned in radians.
+        Default True.
+
+    Returns
+    -------
+    intensity: float or array
+        Intensity of the magnetic vector.
+    inclination : float or array
+        Inclination angle of the magnetic vector.
+        If ``degrees`` is True, then the angle is returned in degree, else it's
+        returned in radians.
+    declination : float or array
+        Declination angle of the magnetic vector.
+        If ``degrees`` is True, then the angle is returned in degrees, else it's
+        returned in radians.
+
+    Examples
+    --------
+    >>> intensity, inc, dec = magnetic_vec_to_ang(1.5, 1.5, -2.12132)
+    >>> print(intensity, inc, dec)
+    2.999999757066657, 44.99999536031435, 45.00000000000001
+    """
+    # Compute the intensity as a norm
+    vectors = np.vstack((magnetic_e, magnetic_n, magnetic_u)).T
+    intensity = np.linalg.norm(vectors, axis=1)
+    # Compute the horizontal component of the magnetic vector
+    horizontal_component = np.array([np.sqrt(magnetic_e**2 + magnetic_n**2)])
+    # Mask the values equal to zero in the horizontal component
+    horizontal_component = np.ma.masked_values(horizontal_component, 0.0)
+    # Calculate the inclination and declination using the mask
+    inclination = np.arctan(-magnetic_u / horizontal_component)
+    declination = np.arcsin(magnetic_e / horizontal_component)
+    # Fill the masked values
+    inclination = inclination.filled(-np.sign(magnetic_u) * np.pi / 2)
+    declination = declination.filled(0)
+    # Convert to degree if needed
+    if degrees:
+        inclination = np.degrees(inclination)
+        declination = np.degrees(declination)
+    if len(intensity) == 1:
+        intensity = intensity[0]
+    return intensity, inclination[0], declination[0]
```

Comment:
We want to return the intensity and two angles as the same type: either all floats or all arrays, so I'm fixing this a little bit. BTW, if we are passing multiple magnetic field vectors, then we expect multiple inclination and declination angles, not only the first element.
```suggestion
    if len(intensity) == 1:
        intensity, = intensity
        inclination, = inclination
        declination, = declination
    return intensity, inclination, declination
```